dali/operators/image/convolution/filter.h

```diff
@@ -0,0 +1,255 @@
+    auto shape = input_shapes[sample_idx];
+    const auto& filter_shape = filter_shapes[sample_idx];
+    for (int dim_idx = 0; dim_idx < filter_shapes.sample_dim(); dim_idx++) {
+      DALI_ENFORCE(
+          filter_shape[dim_idx] > 0,
+          make_string(
+              "Filter of volume zero is not supported, got zero-volume filter for sample of idx ",
+              sample_idx, "."));
+      shape[input_desc.num_seq_dims + dim_idx] -= filter_shape[dim_idx] - 1;
+      DALI_ENFORCE(
+          shape[input_desc.num_seq_dims + dim_idx] > 0,
+          make_string(
+              "Filter for sample of idx ", sample_idx,
+              " is bigger than the sample. This is not allowed if ``mode`` is set to ``valid``."));
+    }
+    output_shapes.set_tensor_shape(sample_idx, shape);
+  }
+  return output_shapes;
+}
+
+// it does not handle `F` extents by the assumption that passing per-frame positional arguments
+// triggers SequenceOperator unfolding
+template <typename In, typename Backend>
+TensorListView<detail::storage_tag_map_t<Backend>, const In, 0> get_fill_values_view(
+    const TensorList<Backend>& fill_values) {
+  DALI_ENFORCE(
+      fill_values.type() == type2id<In>::value,
+      make_string("The padding scalars (third positional argument) must be of the "
+                  "same time as the input samples. Got ",
+                  fill_values.type(), " for pad values while the input samples are of type ",
+                  type2id<In>::value, "."));
+  if (fill_values.sample_dim() == 0) {
+    return view<const In, 0>(fill_values);
+  }
+  const auto& shape = fill_values.shape();
+  int num_samples = shape.num_samples();
+  for (int sample_idx = 0; sample_idx < num_samples; sample_idx++) {
+    DALI_ENFORCE(
+        shape[sample_idx].num_elements() == 1,
+        make_string("Padding values must be scalars, got ", shape[sample_idx].num_elements(),
+                    " elements for a sample of idx ", sample_idx, "."));
+  }
+  auto view_dyn = view<const In>(fill_values);
+  return reshape<0>(view_dyn, uniform_list_shape<0>(num_samples, std::vector<int64_t>{}));
+}
+
+}  // namespace filter
+
+template <typename Backend>
+class Filter : public SequenceOperator<Backend> {
+ public:
+  inline explicit Filter(const OpSpec& spec)
+      : SequenceOperator<Backend>(spec),
+        is_valid_mode_{filter::parse_is_valid_mode(spec.GetArgument<std::string>("mode"))} {
+    spec.TryGetArgument(dtype_, "dtype");
+  }
+
+  DISABLE_COPY_MOVE_ASSIGN(Filter);
+
+ protected:
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+  bool ShouldExpand(const Workspace& ws) override;
+
+  template <typename Out, typename In, typename W>
+  std::unique_ptr<OpImplBase<Backend>> GetFilterImpl(const OpSpec& spec_,
+                                                     const filter::InputDesc& input_desc);
+
+  bool SetupImpl(std::vector<OutputDesc>& output_desc, const Workspace& ws) override {
+    if (!impl_) {
```

Comment:
how about checking that the current impl_ instance matches the previous input and filter type?

Reply:
It makes sense on the one hand, on the other isn't it something that should the exectur do?